Realm/Tests/TransactionTests.m

```diff
@@ -17,10 +17,41 @@
 ////////////////////////////////////////////////////////////////////////////
 
 #import "RLMTestCase.h"
+#import <XCTest/XCTest.h>
 
 @interface TransactionTests : RLMTestCase
 @end
 
+
+
+RLM_ARRAY_TYPE(Breeder)
+
+@interface Breeder : RLMObject
```

Comment:
instead of adding new types `Breeder` and `Dog`, I'd encourage you to reuse an existing model defined in `RLMTestObjects`.